src/main/java/Flights.groovy

```diff
@@ -0,0 +1,152 @@
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+
+class Flights implements FlightList {
+    List<Flight> flights
```

Comment:
IMHO this shouldn't be a class to make instances of but a util class with static methods.
Why always group List<Flight> into another object to process them? Seems an unnecessary step

Do you see another implementation of the FlightList interface? If not , drop the interface.

If you think the unit testing is going to be hard if you don't define an interface, then what are we really testing